develop-docs/self-hosted/experimental/external-kafka.mdx

```diff
@@ -0,0 +1,115 @@
+    "ssl.ca.location": "/path/to/ca.pem",
+    "ssl.certificate.location": "/path/to/client.pem",
+    "ssl.key.location": "/path/to/client.key",
+}
+```
+
+## Snuba
+
+Although Snuba also uses confluent-kafka under the hood, not every configuration option is available. Modify your `docker-compose.yml` file like so:
+
+```yaml
+x-snuba-defaults: &snuba_defaults
+  # ...
+  environment:
+    # ...
+    DEFAULT_BROKERS: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+    KAFKA_SECURITY_PROTOCOL: "plaintext" # Valid options are PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL. SSL is not supported for rust-consumer.
+    KAFKA_SSL_CA_PATH:
+    KAFKA_SSL_CERT_PATH:
+    KAFKA_SSL_KEY_PATH:
+    KAFKA_SASL_MECHANISM: "PLAIN" # Valid options are PLAIN, SCRAM-SHA-256, SCRAM-SHA-512.
+    KAFKA_SASL_USERNAME: "username"
+    KAFKA_SASL_PASSWORD: "password"
+```
+
+If you encounter any failing startup, try to use `consumer` instead of `rust-consumer`.
+
+## Relay
+
+Modify your `relay/config.yml` file like so:
+
+```yaml
+processing:
+  kafka_config:
+    - {name: "bootstrap.servers", value: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"}
+    - {name: "message.max.bytes", value: 50000000} # 50MB
+    - {name: "security.protocol", value: "PLAINTEXT"}
+    - {name: "sasl.mechanism", value: "PLAIN"}
+    - {name: "sasl.username", value: "username"}
+    - {name: "sasl.password", value: "password"}
+    - {name: "ssl.ca.location", value: "/path/to/ca.pem"}
+    - {name: "ssl.certificate.location", value: "/path/to/client.pem"}
+    - {name: "ssl.key.location", value: "/path/to/client.key"}
+```
+
+## Vroom
+
+As of the time of writing, Vroom does not support any kind of authentication.
+
+Modify your `docker-compose.yml` file like so:
+
+```yaml
+vroom:
+  # ...
+  environment:
+    # ...
+    SENTRY_KAFKA_BROKERS_PROFILING: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+    SENTRY_KAFKA_BROKERS_OCCURRENCES: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+    SENTRY_KAFKA_BROKERS_SPANS: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+```
+
+When [vroom#530](https://github.com/getsentry/vroom/pull/530) is merged, you can use authentication. You will need to modify your `docker-compose.yml` file like so:
+
+```yaml
+vroom:
+  # ...
+  environment:
+    # ...
+    SENTRY_KAFKA_BROKERS_PROFILING: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+    SENTRY_KAFKA_BROKERS_OCCURRENCES: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+    SENTRY_KAFKA_BROKERS_SPANS: "kafka-node1:9092,kafka-node2:9092,kafka-node3:9092"
+    SENTRY_KAFKA_SECURITY_PROTOCOL: "plaintext" # Valid options are PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL
+    SENTRY_KAFKA_SSL_CA_PATH: ""
+    SENTRY_KAFKA_SSL_CERT_PATH: ""
+    SENTRY_KAFKA_SSL_KEY_PATH: ""
+    SENTRY_KAFKA_SASL_MECHANISM: "PLAIN" # Valid options are PLAIN, SCRAM-SHA-256, SCRAM-SHA-512.
+    SENTRY_KAFKA_SASL_USERNAME: "username"
+    SENTRY_KAFKA_SASL_PASSWORD: "password"
```

Comment:
```suggestion
    SENTRY_KAFKA_SASL_USERNAME: "<username>"
    SENTRY_KAFKA_SASL_PASSWORD: "<password>"
```

Also, at this point I'm thinking maybe this whole thing should be a `.patch` file they can apply or even lives in the repo and we test for it somehow? Otherwise this would easily become a big hurdle and go out of date/non-functional.

What do you think?

Reply:
But a question: Where should we store these patch files? On the docs? As a static assets?

Reply:
As one of the end users and maintainers of the self-hosted repo, I think that's a question for you to answer best 😃

Reply:
Hahaha. Yeah, I'll come up with something.